src/libAtomVM/atom.c

```diff
@@ -55,20 +55,32 @@ int atom_are_equals(AtomString a, AtomString b)
 
 void atom_write_mfa(char *buf, size_t buf_size, AtomString module, AtomString function, int arity)
 {
+
+    unsigned int arity_num_digits = 0;
```

Comment:
do we have a reason for using `unsigned int`. Most of the times `int` is enough.

Reply:
I guess not! What do you think about using `char` as a 1-byte integer here instead, since the values are always 1, 2, or 3?